Return translated profile descriptions

sgallagher opened the pull request:
This causes profiles to be associated with their owning ModuleStream
when they are added to it so they can look up the appropriate
translations for their description.

Fixes: https://github.com/fedora-modularity/libmodulemd/issues/321

Signed-off-by: Stephen Gallagher <sgallagh@redhat.com>

sgallagher commented:
This change doesn't add tests because I tested it with the tests being added in #319

mmathesius commented on modulemd/v2/modulemd-profile.c:
```diff
@@ -155,7 +160,17 @@ modulemd_profile_get_description (ModulemdProfile *self, const gchar *locale)
+      if (entry != NULL && modulemd_translation_entry_get_profile_description (
+                             entry, self->name) != NULL)
+        return modulemd_translation_entry_get_profile_description (entry,
+                                                                   self->name);
```
(Optional) The above would be a little easier to grok if the flow was along the lines of
```
if (entry != NULL)
  {
    gchar *translation = modulemd_translation_entry_get_profile_description (
                             entry, self->name);
    if (translation != NULL)
      return translation;
  }
```

sgallagher commented:
@mmathesius I made the two changes you suggested, but the CI also indicated that I had introduced a memory leak. It took me a while to figure out, but it was because I was having `Profile.set_owner()` take a reference on the `ModuleStream` that encapsulated it. The result was that I was introducing a refcounting loop and the `ModuleStream` would never be freed. I switched it to just assigning a pointer rather than taking a reference and the memory leak went away.

Please see the fixup commit and confirm the results and I'll squash and merge then.
